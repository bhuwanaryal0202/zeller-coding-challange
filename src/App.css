.app {
  display: flex;
  flex-direction: column;
  justify-content: center;
  font-family: sans-serif;
  width: 100%;
  max-width: 600px;
  margin: 0 auto;
}

h1{
  font-weight: 600;
  font-size: 24px;
  
}

.line {
  border-top: 1px solid #ccc; /* You can adjust the color and thickness as needed */
  margin: 20px 0; /* Adjust the margin to control the space between elements */
}

