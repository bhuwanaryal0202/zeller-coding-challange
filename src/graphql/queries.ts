import { gql } from '@apollo/client';

export const CUSTOMERS_QUERY = gql`      
query ListZellerCustomers {
    listZellerCustomers {
      items {
        email
        id
        name
        role
        __typename
      }
        __typename
    }
  }
`
