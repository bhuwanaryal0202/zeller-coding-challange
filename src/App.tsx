import React,{useState} from 'react';
import {useQuery} from '@apollo/client';
import {CUSTOMERS_QUERY} from './graphql/queries';
import CustomerList from './components/CustomerList';
import './App.css';
import UserTypeSelector from './components/UserTypeSelector';
import client from './apollo';
import { Customer } from './components/types';




function App() {

  const [userType, setUserType] = useState('Admin');

  const {loading, error, data} = useQuery(CUSTOMERS_QUERY, {
    client,
     variables: {userType},
     fetchPolicy: 'cache-first', // Use cached data while fetching new data
  });
  // console.log("Data:", data); 

  if(loading) return <p>Loading...</p>;
  if(error) return <p>Error: {error.message}</p>

  const customers = data.listZellerCustomers.items;
  // console.log("Customers:", customers);

  
   // Filter customers based on the selected userType
   const filteredCustomers = customers.filter((customer: Customer) => customer.role === userType);
   console.log("filteredCustomers:",filteredCustomers  );

  

  return (
    <>
    <div className='app'>
    <div className="line"></div> 
      <h1>User Types</h1>
      <UserTypeSelector 
        userType={userType}
        onUserTypeChange={(newUserType) => setUserType(newUserType)}
      ></UserTypeSelector>
      <div className="line"></div> 
      <h1>Customer List</h1>
      <CustomerList customers={filteredCustomers} />
      <div className="line"></div> 
    </div>
    </>
  );
}

export default App;
