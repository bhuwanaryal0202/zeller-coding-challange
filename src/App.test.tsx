/**
 * @format
 */
import App from './App';
import { MockedProvider } from '@apollo/client/testing';
import { CUSTOMERS_QUERY } from './graphql/queries';
import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';

const adminMocks = [
    {
        request: {
            query: CUSTOMERS_QUERY,
            variables: { userType: 'Admin' },
        },
        result: {
            data: {
                customers: [
                    {
                        id: '1',
                        name: 'John Doe',
                        email: 'johndoe@example.com',
                    },
                    {
                        id: '2',
                        name: 'Jane Doe',
                        email: 'janedoe@example.com',
                    },
                ],
            },
        },
    },
];

const managerMocks = [
    {
        request: {
            query: CUSTOMERS_QUERY,
            variables: { userType: 'Manager' },
        },
        result: {
            data: {
                customers: [
                    {
                        id: '1',
                        name: 'John Doe',
                        email: 'johndoe@example.com',
                    },
                    {
                        id: '2',
                        name: 'Jane Doe',
                        email: 'janedoe@example.com',
                    },
                ],
            },
        },
    },
];

it('renders admin correctly', async () => {
    let tree: renderer.ReactTestRenderer | null = null; // Fix for Problem 1
    await act(async () => {
        tree = renderer.create(
            <MockedProvider mocks={adminMocks} addTypename={false}>
                <App />
            </MockedProvider>,
        );
    });
    expect(tree!.toJSON()).toMatchSnapshot(); // Fix for Problem 2
});


it('renders manager correctly', async () => {
    let tree: renderer.ReactTestRenderer | null = null; // Fix for Problem 1
    await act(async () => {
        tree = renderer.create(
            <MockedProvider mocks={managerMocks} addTypename={false}>
                <App />
            </MockedProvider>,
        );
    });
    expect(tree!.toJSON()).toMatchSnapshot(); // Fix for Problem 2
});

