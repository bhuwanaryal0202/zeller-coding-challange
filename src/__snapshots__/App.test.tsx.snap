// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders admin correctly 1`] = `
<p>
  Loading...
</p>
`;

exports[`renders correctly 1`] = `
<p>
  Loading...
</p>
`;

exports[`renders correctly 2`] = `
<p>
  Loading...
</p>
`;

exports[`renders manager correctly 1`] = `
<p>
  Loading...
</p>
`;
